homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -150,21 +153,25 @@ class DeviceConnectivity(SurePetcareBinarySensor):
     def __init__(
         self,
         _id: int,
-        spc: SurePetcareAPI,
+        coordinator: DataUpdateCoordinator,
     ) -> None:
         """Initialize a Sure Petcare Device."""
-        super().__init__(_id, spc, DEVICE_CLASS_CONNECTIVITY)
+        super().__init__(_id, coordinator, DEVICE_CLASS_CONNECTIVITY)
         self._attr_name = f"{self.name}_connectivity"
         self._attr_unique_id = (
-            f"{self._spc.states[self._id].household_id}-{self._id}-connectivity"
+            f"{self.coordinator.data[self._id].household_id}-{self._id}-connectivity"
         )
 
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return super().available and bool(self._attr_is_on)
+
     @callback
-    def _async_update(self) -> None:
-        """Get the latest data and update the state."""
-        surepy_entity = self._spc.states[self._id]
+    def _update_attr(self):
+        surepy_entity = self.coordinator.data[self._id]
         state = surepy_entity.raw_data()["status"]
-        self._attr_is_on = self._attr_available = bool(state)
+        self._attr_is_on = bool(state)
```

Comment:
you could set self._attr_available here as well and avoid property available.

Reply:
Then I do not get the available state from the DataUpdateCoordinator?

Reply:
You are right.